test/wasm2js/atomic_fence.2asm.js

```diff
@@ -55,10 +57,7 @@ function asmFunc(env, buffer) {
  };
 }
 
-var memasmFunc = new ArrayBuffer(1507328);
-var bufferView = new Uint8Array(memasmFunc);
+var bufferView;
```

Comment:
can we not remove this `bufferView` entirely, and just use `HEAP8` which is guaranteed to exist?

Reply:
See my `TODO: We should probably just share a single HEAPU8 var.` above. 

I was thinking that could be a followup.     Keeping this change slightly more limited.

Reply:
This does make it harder to review, though, see the comment above with `unconditional` - it looks like the current PR is not emitting `bufferView` properly, but I may be missing something?

Reply:
If we start to share HEAPU8 with the containing scope is that going to interfere with the emscripten-side that also defines a HEAPU8 itself?   I mean that should always point to the same thing anyway so maybe its fine?

Reply:
Yes it looks like the wasm2js output shares a scope with the rest of emscripten, but `asmFunc` currently creates it own function-local versions of HEAP8, etc.   This means that in emscripten output there are two copies of all these views today.   

We should possibly merge those two and (i.e. have the output HEAP8 be the same as the inner HEAP8), but I'd rather not worry about that as part of this change.